lib/puppet/provider/keycloak_realm/kcadm.rb

```diff
@@ -244,9 +269,13 @@ def flush
         if self.class.browser_security_headers.include?(property) && !data.key?('browserSecurityHeaders')
           data['browserSecurityHeaders'] = {}
         end
+        if self.class.smtp_server_properties.include?(property)
```

Comment:
Same as other comment:

This should be self.class.smtp_server_properties.include?(property) && !data.key?('smtpServer') and then if this is true it adds the hash data['smtpServer'] = {} so that you are only adding the key if SMTP server settings exist. Then this current assignemtn gets moved to line 279.

Reply:
I implemented this as requested but (fortunately) noticed a glitch as I was writing acceptance tests. If done as you suggested changing the property values will actually destroy all the properties that do not change. Suppose I run this code to create a new realm:
```
keycloak_realm { 'foobar':
  ensure => 'present',
  smtp_server_host => 'server.example.org',
  smtp_server_port => '25',
}
```
Then I change  _smtp_server_host_  to _foobar.example.org_ what I end up with in the realm SMTP settings is this:

```
"smtpServer" : {
    "host" : "foobar.example.org"
}
```
So the "port" is gone from the realm.

I ran out of time today and could not debug this further, but the original/current version I have here does work.

Not sure if this affects the flush implementation, but when a new realm is created using kcadm an empty smtpServer hash is always present:

```
$ /opt/keycloak/bin/kcadm-wrapper.sh create realms -s realm=foobar -s enabled=true -o|grep smtpServer
Logging into http://localhost:8080/auth as user admin of realm master
  "smtpServer" : { },
```
Any suggestions would of course be welcome.

Reply:
The reason you lost items is because the line 275 condition only looks at properties if they are defined in `@property_flush` which gets populated only with changed things.  It might be possible to work around this issue by removing commented part on line 275 to also check `resource` which would allow you to make my suggested changes and not end up with some items missing if they don't change.  The acceptance tests should tell if you removing that comment breaks things or not.

I'll add some suggested changes

Reply:
Thanks, I'll see if that breaks things and report back.